src/debugger/channel/AdapterChannel.js

```diff
@@ -10,20 +10,113 @@
+    this.listenOnFile(this._processPrepackMessage.bind(this));
   }
   _ioWrapper: FileIOWrapper;
+  _messageHandler: MessageHandler;
+  _queue: Queue;
+  _pendingRequestCallbacks: { [number]: (string) => void };
+  _prepackWaiting: boolean;
+  _eventEmitter: EventEmitter;
+
+  // Error handler for errors in files from the adapter channel
+  _handleFileReadError(err: ?ErrnoError) {
+    console.error(err);
+    process.exit(1);
+  }
+
+  _processPrepackMessage(message: string) {
+    let parts = message.split(" ");
+    let requestID = parseInt(parts[0], 10);
+    invariant(!isNaN(requestID));
+    let prefix = parts[1];
+    if (prefix === DebugMessage.PREPACK_READY_RESPONSE) {
+      this._prepackWaiting = true;
+      this._eventEmitter.emit(DebugMessage.PREPACK_READY_RESPONSE);
+      this.trySendNextRequest();
+    } else if (prefix === DebugMessage.BREAKPOINT_ADD_ACKNOWLEDGE) {
+      this._eventEmitter.emit(DebugMessage.BREAKPOINT_ADD_ACKNOWLEDGE, requestID, message);
+      // Prepack acknowledged adding a breakpoint
+      this._prepackWaiting = true;
+      this._processRequestCallback(requestID, message);
+      this.trySendNextRequest();
+    } else if (prefix === DebugMessage.BREAKPOINT_STOPPED_RESPONSE) {
+      this._eventEmitter.emit(DebugMessage.BREAKPOINT_STOPPED_RESPONSE, parts.slice(2).join(" "));
+      // Prepack stopped on a breakpoint
+      this._prepackWaiting = true;
+      this._processRequestCallback(requestID, message);
+      this.trySendNextRequest();
+    }
+  }
+
+  // Check to see if the next request to Prepack can be sent and send it if so
+  trySendNextRequest(): boolean {
+    // check to see if Prepack is ready to accept another request
+    if (!this._prepackWaiting) return false;
+    // check that there is a message to send
+    if (this._queue.isEmpty()) return false;
+    let request = this._queue.dequeue();
+    this.listenOnFile(this._processPrepackMessage.bind(this));
+    this.writeOut(request);
+    this._prepackWaiting = false;
+    return true;
+  }
+
+  _addRequestCallback(requestID: number, callback: string => void) {
+    invariant(!(requestID in this._pendingRequestCallbacks), "Request ID already exists in pending requests");
+    this._pendingRequestCallbacks[requestID] = callback;
+  }
+
+  _processRequestCallback(requestID: number, message: string) {
+    invariant(
+      requestID in this._pendingRequestCallbacks,
+      "Request ID does not exist in pending requests: " + requestID
+    );
+    let callback = this._pendingRequestCallbacks[requestID];
+    callback(message);
+  }
+
+  queueContinueRequest(requestID: number, callback: string => void) {
+    this._queue.enqueue(this._messageHandler.formatContinueRequest(requestID));
+    this.trySendNextRequest();
+    this._addRequestCallback(requestID, callback);
+  }
+
+  queueSetBreakpointsRequest(requestID: number, breakpoints: Array<BreakpointArguments>, callback: string => void) {
```

Comment:
Just call it "setBreakpoints()". The fact it is queued is an implementation details.